### ISSUE TITLE
Make a disabled button and action more accessible

### ISSUE THREAD

**Issue by cont-amibabkin** [Esri/calcite-components#13296]
**monday.com sync:** #10526165688

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Action, Button

### Summary

When disabled, the button and action have two attributes on the calcite tag: disabled and aria-disabled. The button in shadow root has the disabled attribute. We believe the aria-disabled attribute is irrelevant in this case, as users using assistive technology won't hear the button's status, which visually indicates its deactivation.

We have a use case where a disabled action needs to be accessible to assistive technology users (we have a similar use case for a button). This action has a tooltip that explains why the button was disabled. Currently, screen reader users can't access this action in either Browse mode or Focus mode. Therefore, we want to have a disabled button and action version that only uses the aria-disabled attribute on the button in shadow root.

It would be nice to have two options for the disabled button: the one that exists now and the one we offer.

### Actual Behavior

The disabled button and action are currently not available to assistive technology users.

### Expected Behavior

Assistive technology users have access to a disabled button with a tooltip:

<img width="721" height="377" alt="Image" src="https://github.com/user-attachments/assets/fcbbcbf5-6d90-40da-9ebd-d0de9b93b980" />

### Reproduction Sample

https://codepen.io/Andrey-B-the-animator/pen/YPqKQwx

### Reproduction Steps

1. Activate the screen reader.
2. Ensure the screen reader user does not have access to the disabled calcite-action.
3. Ensure the screen reader user has access to the disabled custom button.

### Reproduction Version

3.3.3

### Working W3C Example/Tutorial

https://advancedbytez.com/make-disabled-buttons-accessible/#1_Use_aria-disabled

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by geospatialem** [comment]
@cont-amibabkin Calcite follows native `disabled` functionality, however the above can be achieved preventing the default behavior and some context provided both visually and to assistive technologies.

More details in https://github.com/Esri/calcite-design-system/issues/12547#issuecomment-3092487419, which includes https://codepen.io/geospatialem/pen/xbwZrNV on some options to provide context visually and to assistive technologies.

Closing the above as unplanned.
